mne/fixes.py

```diff
@@ -1219,3 +1219,19 @@ def __repr__(self):
     # __getstate__ and __setstate__ are omitted because they only contain
     # conditionals that are not satisfied by our objects (e.g.,
     # ``if type(self).__module__.startswith('sklearn.')``.
+
+
+###############################################################################
+# NumPy einsum backward compat (allow "optimize" arg and fix 1.14.0 bug)
+# XXX eventually we should hand-tune our `einsum` calls given our array sizes!
+
+_has_optimize = (LooseVersion(np.__version__) >= '1.12')
+
+
+def einsum(*args, **kwargs):
+    if 'optimize' in kwargs:
+        if not _has_optimize:
+            kwargs.pop('optimize')
```

Comment:
FYI this is useful to avoid the speed regression mentioned in:

https://github.com/numpy/numpy/issues/10357

Eventually, as noted here, we should probably manually optimize the path (though it shouldn't matter much because our `einsum`s are generally small.